collector/linux_threads.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2015 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !nothreads
+
+package collector
+
+import (
+	"fmt"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/procfs"
+	"io/ioutil"
+	"strconv"
+	"strings"
+)
+
+type threadsCollector struct {
+	threadAlloc *prometheus.Desc
+	threadMax *prometheus.Desc
+	procsState   *prometheus.Desc
+}
+
+func init() {
+	registerCollector("threads", defaultDisabled, NewThreadsCollector)
+}
+
+func NewThreadsCollector() (Collector, error) {
+	return &threadsCollector{
+		threadAlloc: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, "", "threads"),
+			"Allocated threads in system",
+			nil, nil,
+		),
+		threadMax: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, "threads", "max"),
+			"Limit of threads in the system",
+			nil, nil,
+		),
+		procsState: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, "", "procs_state"),
+			"Number of processes in each state.",
+			[]string{"state"}, nil,
+		),
+	}, nil
+}
+func (t *threadsCollector) Update(ch chan<- prometheus.Metric) error {
+	states ,threads, err := getAllocatedThreads()
```

Comment:
formatting issue, can you please run gofmt?
Also, we should give the function a better name. getProcStats() or something like that.